test/functorch/test_aotdispatch.py

```diff
@@ -2071,10 +2070,7 @@ def f(a, b):
             torch.ones(3, 3, requires_grad=True),
             torch.ones(3, 3, requires_grad=True),
         ]
-        with self.assertRaisesRegex(
-            AssertionError, "input that requires_grad and was mutated in the backward"
-        ):
-            self.verify_aot_autograd(f, inp_grad, test_mutation=True)
+        self.verify_aot_autograd(f, inp_grad, test_mutation=True)
```

Comment:
nit: can you add a quick comment in the test explaining that we can properly handle keeping the backward mutation in the graph in this test because the backward is running under no_grad? (someone is gonna look at this test in a year and have to squint really hard to figure that out)